package/sbp_fileio_daemon/sbp_fileio_daemon/src/sbp_fileio.c

```diff
@@ -597,13 +983,21 @@ static void read_cb(u16 sender_id, u8 len, u8 msg_[], void *context)
       }
     }
 
-    read_result = fread(&reply->contents, 1, readlen, the_file.fp);
-    if (read_result != readlen) PK_LOG_ANNO(LOG_ERR, "fread failed: %s", strerror(errno));
-
-    finalize_non_cached(&the_file);
+    if (seek_result == 0) {
+      wq_run_when_empty(NESTED_FN(void, (), {
```

Comment:
I can understand placing priority on writes for a large transfer scenario, but I'm wondering if there is a chance that read request servicing could be starved out in situations like ndb read/writes where the use case is less throughput focused and needs low latency performance. Is there any chance of this becoming an issue or am I interpreting incorrectly?

Reply:
There is probably some meat to your theory that interleaved write/read/delete's all blocking on writes might cause issues.  I think the long term solution is to move all IO operations into the thread.  Added https://swift-nav.atlassian.net/browse/ESD-1124 to handle this.